dataprep/eda/create_report/formatter.py

```diff
@@ -0,0 +1,188 @@
+"""
+    This module implements the formatting
+    for create_report(df) function.
+"""
+from typing import Optional, Dict, List, Any
+
+import pandas as pd
+import numpy as np
+from tqdm import tqdm
+from bokeh.embed import components
+from bokeh.plotting import Figure
+from bokeh.models import Title
+from ..basic import compute, render
+from ..missing import compute_missing, render_missing
+from ..correlation import compute_correlation, render_correlation
+from ..dtypes import is_dtype, Continuous, detect_dtype
+from ..intermediate import Intermediate
+
+
+def format_report(df: pd.DataFrame, mode: Optional[str]) -> Dict[str, Any]:
+    """
+    Format the data and figures needed by report
+    """
+
+    comps: Dict[str, Any] = {}
+    if mode == "basic":
+        comps = format_basic(df, comps)
+    # elif mode == "full":
+    #     comps = format_full(df, comps)
+    # elif mode == "minimal":
+    #     comps = format_mini(df, comps)
+
+    return comps
+
+
+def format_basic(df: pd.DataFrame, comps: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Format basic version.
+    """
+
+    # pylint: disable=too-many-locals, too-many-statements
+    with tqdm(total=100) as pbar:
+        # plot(df)
+        itmdt = compute(df)
```

Comment:
Is it possible to avoid all the computations for the plots in `plot(df)` and instead compute just the stats?

Reply:
I think it's possible but not very necessary because part of stat data is from `compute_overview` function.